components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java

```diff
@@ -74,6 +89,11 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
                 return AuthenticatorFlowStatus.INCOMPLETE;
             } else {
                 try {
+                    boolean isBasicAuthenticator = this instanceof BasicAuthenticator;
```

Comment:
We can't have a dependency for BasicAuthenticator in here. Authenticator specific logic should be handled from the authenticator, not in the AbstractApplicationAuthenticator. You can improve this method to easily implement this functionality in BasicAuthenticator